### PR TITLE
Nfdiv 3988 urgent flag general application

### PR DESCRIPTION
**Pull request by pallavijustice**
### Change description ###
On the general application event for Solicitors, Service has requested for an urgent box for the Solicitors to tick to state whether or not the application is urgent.

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-3988

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerUpdateGeneralApplication.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        new PageBuilder(configBuilder
+            .event(CASEWORKER_UPDATE_GENERAL_APPLICATION)
+            .forStates(POST_ISSUE_STATES)
+            .name("Update general application")
+            .description("Update general application")
+            .showEventNotes()
+            .grant(CREATE_READ_UPDATE, SUPER_USER, CASE_WORKER)
+            .grantHistoryOnly(
+                SOLICITOR,
+                CASE_WORKER
```
Comment (FaisalMoJ) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerUpdateGeneralApplication.java:35]: I believe this line will prevent it for solicitor to show this on the event dropdown

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerUpdateGeneralApplication.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+@Slf4j
+public class CaseworkerUpdateGeneralApplication implements CCDConfig<CaseData, State, UserRole> {
+    public static final String CASEWORKER_UPDATE_GENERAL_APPLICATION = "caseworker-update-general-application";
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
+            .event(CASEWORKER_UPDATE_GENERAL_APPLICATION)
+            .forStates(POST_ISSUE_STATES)
+            .name("Update general application")
```
Comment (FaisalMoJ) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerUpdateGeneralApplication.java:29]: May be introduce constant for this as it is used a few times

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorGeneralApplication.java [hunk 3]
```diff
@@ -119,6 +122,15 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(final CaseDet
 
         final GeneralApplication generalApplication = data.getGeneralApplication();
 
+        if (generalApplication.getGeneralApplicationUrgentCase() == YesOrNo.YES) {
+            if (isNull(generalApplication.getGeneralApplicationUrgentCaseReason())
+                || isEmpty(generalApplication.getGeneralApplicationUrgentCaseReason())) {
```
Comment (FaisalMoJ) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorGeneralApplication.java:127]: Replace this with isNullOrEmpty()?

**Review by FaisalMoJ — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralApplication.java [hunk 4]
```diff
@@ -52,4 +53,15 @@ public class GeneralApplication {
...
     private FeeDetails generalApplicationFee = new FeeDetails();
+
+    @CCD(
+        label = "Is this an Urgent Application?"
+    )
+    private YesOrNo generalApplicationUrgentCase;
+
+    @CCD(
+        label = "How does this Qualify as an urgent Application?",
+        typeOverride = TextArea
```
[the remark this answers is not in the document]
Comment (pallavijustice) [reply src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralApplication.java:64]: I can add the hint parameter to this field. However that was not currently in the AC.
The urgent flag and reason was expected to behave similar to general referral where we don't currently use Hint.

[the remark this answers is not in the document]
Comment (FaisalMoJ) [reply src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralApplication.java:64]: Also i noticed a mix of words starting with upper and lower letters for label. In most cases i have seen only the first word as initcap

[the remark this answers is not in the document]
Comment (pallavijustice) [reply src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralApplication.java:64]: I can check with Fubara to confirm but the ACs within the ticket have them in the same way. Did not make sense to me either but noticed the same thing on some other events as well.

**Review by FaisalMoJ — APPROVED** [review]

**Pull request merged**